nix/modules/go.nix

```diff
@@ -14,7 +14,7 @@
       packages = {
         goWrapper =
           let
-            go = pkgs.go_1_21;
+            go = pkgs.go_1_23;
```

Comment:
Older Go version no longer available in the nixpkgs we're using

Reply:
I assume there aren't any breaking changes between these minor versions? Also, why this version and not just the default `pkgs.go`?

Reply:
This is a good point. I believe it was pinned because at some point it broke but it's not like the CI won't tell us and we can pin again if we have to. I'll update this to use be `go`